### PR TITLE
Added another overloads of blobFromImage functions 

### PR DESCRIPTION
**Pull request by arrybn**
### This pullrequest changes
Added blobFromImage functions which can crop input from the center and subtract mean

### CODE REVIEW

File: modules/dnn/src/dnn.cpp [hunk 1]
```diff
@@ -154,6 +155,38 @@ Mat blobFromImages(const std::vector<Mat>& images, double scalefactor, bool swap
...
+            size = imgSize;
+        float resizeFactor = std::max(size.width / (float)imgSize.width,
+                                      size.height / (float)imgSize.height);
+        images[i].convertTo(images[i], CV_32F);
+        resize(images[i], images[i], Size(), resizeFactor, resizeFactor);
+        Rect crop(Point(0.5 * (images[i].cols - size.width),
+                        0.5 * (images[i].rows - size.height)),
+                  size);
+        Scalar mean = mean_;
+        if (swapRB)
```
Comment (dkurt) [inline modules/dnn/src/dnn.cpp]: It seems to me that with `swapRB == true` we'll do it twice. In this function and in origin `blobFromImages`.

Comment (arrybn) [inline modules/dnn/src/dnn.cpp]: here swap is only for mean values Scalar, not for input image. there's no double swapping

Comment (dkurt) [inline modules/dnn/src/dnn.cpp]: Yeah, thanks, I see.

**Comment by dkurt** [comment]
:+1:

File: modules/dnn/include/opencv2/dnn/dnn.hpp [hunk 2]
```diff
@@ -598,9 +598,55 @@ namespace dnn //! This namespace is used for dnn module functionlaity.
...
+     *  @param mean scalar with mean values which are subtracted from channels. Values are intended
+     *  to be in (mean-R, mean-G, mean-B) order if @p image has BGR ordering and @p swapRB is true.
+     *  @param scalefactor multiplier for @p image values.
+     *  @param swapRB flag which indicates that swap first and last channels
+     *  in 3-channel image is necessary.
+     *  @details input image is resized so one side after resize is equal to corresponing
+     *  dimension in @p size and another one is equal or larger. Then, crop from the center is performed.
+     *  @returns 4-dimansional Mat with NCHW dimensions order.
+     */
+    CV_EXPORTS_W Mat blobFromImage(const Mat& image, const Size& size,
```
Comment (vpisarev) [inline modules/dnn/include/opencv2/dnn/dnn.hpp]: can we just make the `size` parameter optional too? (`const Size& size=Size()`) and reduce the number of overloaded methods?

Comment (arrybn) [inline modules/dnn/include/opencv2/dnn/dnn.hpp]: Removed 2 overloads

**Comment by vpisarev** [comment]
:+1:

**Pull request merged**